csharp/src/Apache.Arrow/Arrays/BooleanArray.cs

```diff
@@ -153,17 +184,25 @@ private void CheckIndex(int index)
                 new[] { nullBitmapBuffer, valueBuffer }))
         { }
 
-        public BooleanArray(ArrayData data) 
+        public BooleanArray(ArrayData data)
             : base(data)
         {
             data.EnsureDataType(ArrowTypeId.Boolean);
         }
 
         public override void Accept(IArrowArrayVisitor visitor) => Accept(this, visitor);
 
+        [Obsolete("GetBoolean does not support null values. Use GetValue instead (which this method invokes internally).")]
         public bool GetBoolean(int index)
         {
-            return BitUtility.GetBit(Values, index);
+            return GetValue(index).GetValueOrDefault();
```

Comment:
Even though this is Obsolete, I still think it should act appropriately. I think it should call `GetValue(index).Value`. If `null` is at `index` that will create an exception, which informs the developer they have a bug in their code. That is better than silently lying to them and returning `false`.

Reply:
Agree. Changed.

Reply:
Changed in the test code uses of `GetValueOrDefault` to `Value`, but think this one should remain `GetValueOrDefault`. Existing code is likely to break otherwise. I've seen it already with the tests. Current code written against `GetBoolean` will not be expecting an exception, and it's easy enough for a null to get introduced somewhere else that eventually causes `GetBoolean` throw.